File: Admin-Super-Admin/admin/src/types/ticket.ts
export type TicketStatus = "Open" | "In Progress" | "Resolved"
export type TicketCategory = "Technical Issue" | "Payment Issue" | "Ad Issue" | "Other"

export interface Message {
  id: string
  sender: "Admin" | "Super Admin"
  content: string
  timestamp: string
  attachments?: {
    name: string
    url: string
    type: string
  }[]
}

export interface Ticket {
  id: string
  subject: string
  description: string
  category: TicketCategory
  status: TicketStatus
  createdAt: string
  updatedAt: string
  messages: Message[]
}

File: Admin-Super-Admin/backend/src/main/java/org/jackfruit/keliri/model/Ticket.java
package org.jackfruit.keliri.model;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.time.Instant;

@Document(collection = "tickets")
public class Ticket {

    @Id
    private String id;

    private String adminId;

    private String subject;
    private String category; // TECHNICAL / PAYMENT / AD / OTHER

    private String status; // OPEN / IN_PROGRESS / RESOLVED

    private Instant createdAt;
    private Instant updatedAt;

    public Ticket() {}

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getAdminId() {
        return adminId;
    }

    public void setAdminId(String adminId) {
        this.adminId = adminId;
    }

    public String getSubject() {
        return subject;
    }

    public void setSubject(String subject) {
        this.subject = subject;
    }

    public String getCategory() {
        return category;
    }

    public void setCategory(String category) {
        this.category = category;
    }

    public String getStatus() {
        return status;
    }

    public void setStatus(String status) {
        this.status = status;
    }

    public Instant getCreatedAt() {
        return createdAt;
    }

    public void setCreatedAt(Instant createdAt) {
        this.createdAt = createdAt;
    }

    public Instant getUpdatedAt() {
        return updatedAt;
    }

    public void setUpdatedAt(Instant updatedAt) {
        this.updatedAt = updatedAt;
    }
}
